packages/icon/src/react/index.js

```diff
@@ -1,40 +1,40 @@
-import glamorous from 'glamorous'
+import * as glamor from 'glamor'
 import PropTypes from 'prop-types'
 import React from 'react'
 
+import filterReactProps from '@pluralsight/ps-design-system-filter-react-props'
+
 import icons from '../js/icons'
 
 import css from '../css'
 import * as vars from '../vars'
 
-const IconContainer = glamorous.div(
-  css['.psds-icon'],
-  ({ size }) => css[`.psds-icon--size-${size}`],
-  ({ color }) => ({
-    '> svg': color
-      ? css[`.psds-icon--color-${color} > svg`]
-      : css['.psds-icon > svg']
-  }),
-  ({ css: propsCss }) => propsCss
-)
+const style = {
+  icon: ({ css: propCSS = {}, color, size }) =>
+    glamor.css({
+      ...css['.psds-icon'],
+      ...css[`.psds-icon--size-${size}`],
 
-const rmNonHtmlProps = props => {
-  const { id, ...rest } = props
-  return rest
-}
+      '> svg': color
+        ? css[`.psds-icon--color-${color} > svg`]
+        : css['.psds-icon > svg'],
 
-const Icon = props => {
-  return (
-    <IconContainer {...rmNonHtmlProps(props)}>
-      {props.children || icons[props.id](React)}
-    </IconContainer>
-  )
+      // NOTE: this is a holdover from the old glamorous api,
```

Comment:
still a desire to remove?  Or did you find a usecase where it's still required to do what most users need to do with icon?

Reply:
no i removed it